frontend/src/lib/WorkflowParser.ts

```diff
@@ -231,30 +240,64 @@ export default class WorkflowParser {
   // inputs/outputs artifacts if any, while looking out for any missing link in the chain to
   // the node's inputs/outputs.
   public static getNodeInputOutputArtifacts(
-    workflow?: Workflow,
+    workflow?: any,
     nodeId?: string,
   ): Record<'inputArtifacts' | 'outputArtifacts', Array<KeyValue<S3Artifact>>> {
     type ParamList = Array<KeyValue<S3Artifact>>;
     let inputArtifacts: ParamList = [];
     let outputArtifacts: ParamList = [];
-    if (
-      !nodeId ||
-      !workflow ||
-      !workflow.status ||
-      !workflow.status.nodes ||
-      !workflow.status.nodes[nodeId]
-    ) {
+
+    if (!workflow || !workflow.metadata || !workflow.status || !workflow.metadata.annotations)
       return { inputArtifacts, outputArtifacts };
-    }
 
-    const { inputs, outputs } = workflow.status.nodes[nodeId];
-    if (!!inputs && !!inputs.artifacts) {
-      inputArtifacts = inputs.artifacts.map(({ name, s3 }) => [name, s3]);
-    }
-    if (!!outputs && !!outputs.artifacts) {
-      outputArtifacts = outputs.artifacts.map(({ name, s3 }) => [name, s3]);
+    // Get the task name that corresponds to the nodeId
+    let taskName = "";
+    for (const taskRunId of Object.getOwnPropertyNames(workflow.status.taskRuns)) {
```

Comment:
Create a fallback here as well, it will break when the pod/pipeline is not yet running since the taskRun is not yet created.

Reply:
Would returning if !workflow.status.taskRuns cover both these cases, since taskRun status is only available when pod/pipeline is running?

Reply:
As long as it won't break the page it should be good. The ui is constantly pulling new status every 5-10s